pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -107,7 +107,7 @@ func BuildSidecarVirtualHostsFromConfigAndRegistry(
 	for fqdn := range missing {
 		svc := serviceRegistry[fqdn]
 		for _, port := range svc.Ports {
-			if port.Protocol.IsHTTP() {
+			if port.Protocol.IsHTTP() || (util.IsIstioVersionGE13(node) && port.Protocol.IsUnsupported()) {
```

Comment:
not sure I understand why we check for Unsupported here rather than Auto? When will this be auto and when will it be unsupported

Reply:
`Auto` is used internal. `Unsupported` is user interface.

Reply:
I thought from the design doc users could specify auto in Sidecar API?

Reply:
Yes. But as earlier comment in the pr, we don't expose `auto` to users at the beginning.